### PR TITLE
Search upwards if the tasks directory isn't found locally.

### LINKED ISSUE #1211: Use require() when loading NPM tasks
I wrote a library using grunt to build itself. Let's call it L.
I wrote a project using grunt to build itself, and using L as a dependency. Let's call it P.
Both L and P are using the same grunt plugin. Let's call it G.

Because npm does install a library just once in a node_modules tree, here is the hierarchy of the P project after running `npm install`:

```
./P
./P/node_modules
./P/node_modules/G
./P/node_modules/L
```

But there is no `./P/node_modules/L/node_modules/G` and that makes grunt fail to build L.

Why is grunt searching for a `./node_modules/<plugin>/package.json` file instead of just requiring it like any other node module?

### PR DESCRIPTION
**Pull request by rbelouin**
Fixes #1211.

It actually fix some cases that aren't handled by #1229.

### CODE REVIEW

File: lib/grunt/task.js [hunk 1]
```diff
@@ -380,7 +380,14 @@ task.loadTasks = function(tasksdir) {
 task.loadNpmTasks = function(name) {
   loadTasksMessage('"' + name + '" local Npm module');
   var root = path.resolve('node_modules');
-  var pkgfile = path.join(root, name, 'package.json');
+  var pkgdir = path.join(root, name);
+
+  // If the package directory is not found, try to find upwards
+  if(!grunt.file.exists(pkgdir)) {
+    pkgdir = grunt.file.findup('node_modules/' + name);
```
Comment (sindresorhus) [inline lib/grunt/task.js]: You need to use `path.join` here.

Comment (gyandeeps) [inline lib/grunt/task.js]: I thought `findup` was deprecated?

**Comment by rbelouin** [comment]
It seems you're right @gyandeeps, `grunt.file.findup()` looks deprecated. I don't know if it was the case last October.

I will fix this using `findup-sync` as it's already included in the `package.json` file.

**Comment by shama** [comment]
Closing as a duplicate of https://github.com/gruntjs/grunt/pull/1229 Thank you for the PR though!

**Pull request closed without merging**